From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 12 May 2019 13:58:04 +0100
Subject: x86/msr-index: Remove dependency on <linux/bits.h>
Forwarded: not-needed

The x86 <asm/msr-index.h> header is (mistakenly) exported as a UAPI
header in 4.9.  Adding the inclusion of <linux/bits.h> stops that
working entirely as that header isn't exported.

This has been resolved upstream by un-exporting <asm/msr-index.h>.
However, to avoid build regressions for user-space in stretch, we
should keep it working as a UAPI header.

Make the inclusion of <linux/bits.h> conditional on __KERNEL__.
Although the bit macros defined with BIT() and BIT_ULL() won't work in
user-space, that's not a regression.
---
--- a/arch/x86/include/asm/msr-index.h
+++ b/arch/x86/include/asm/msr-index.h
@@ -1,7 +1,9 @@
 #ifndef _ASM_X86_MSR_INDEX_H
 #define _ASM_X86_MSR_INDEX_H
 
+#ifdef __KERNEL__
 #include <linux/bits.h>
+#endif
 
 /*
  * CPU model specific register (MSR) numbers.
